external/llvm-project/mlir/lib/ExecutionEngine/CMakeLists.txt

```diff
@@ -384,6 +384,9 @@ if(LLVM_ENABLE_PIC)
       intrinsics_gen
     )
 
+    set_property(TARGET obj.MLIRRocmExecutionEngineUtils
```

Comment:
Hack.  The one file in this library, RocmSystemDetect.cpp, was not getting the -D__HIP_PLATFORM_AMD__=1 that it needed to compile.  Setting this property here is the smaller cleaner alternative to using the old AddMLIR.cmake instead of the upstream one.  I should probably pull it from hip::host instead of hard-coding.

It seems to indicate an issue with how properties are propagated with the recent cmake changes that prepend "obj." to target names, but I haven't tracked it further than that yet.

Reply:
Could it be the case that you're missing a `target_link_libraries` here?

Reply:
target_link_libraries is there, target_compile_definitions is there, but somewhere the property is not communicated to the right place.

Reply:
I think the `target_link_libraries` is going to a different target. Pay attention to which library the system name stuff is part of.

Reply:
I think you're right.  It seems I need target_link_libraries for both MLIRRocmExecutionEngineUtils and obj.MLIRRocmExecutionEngineUtils -- one to get the -D__HIP_PLATFORM_AMD__ for compilation and one to get HIP stuff for linking.  With those, I don't need this hack.  I'll see how small a change I can get away with and push it.